asyn/devEpics/devAsynXXXArray.cpp

```diff
@@ -0,0 +1,573 @@
+                    INIT_FUNC, GET_INFO_FUNC, PROC_FUNC, QRCB_FUNC, INTCB_FUNC,            \
+                    SIGNED_TYPE, UNSIGNED_TYPE, IS_OUTPUT)                                 \
+static long GET_INFO_FUNC(int cmd, dbCommon *pr, IOSCANPVT *iopvt) {                       \
+    devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *pObj =                         \
+        (devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *) pr->dpvt;               \
+    return pObj->getIoIntInfo(cmd, iopvt);                                                 \
+}                                                                                          \
+static long PROC_FUNC(dbCommon *pr) {                                                      \
+    devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *pObj =                         \
+        (devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *) pr->dpvt;               \
+    return pObj->process();                                                                \
+}                                                                                          \
+static void QRCB_FUNC(asynUser *pasynUser) {                                               \
+    devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *pObj =                         \
+        (devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *) pasynUser->userPvt;     \
+    pObj->queueRequestCallback();                                                          \
+}                                                                                          \
+static void INTCB_FUNC(void *drvPvt, asynUser *pasynUser, EPICS_TYPE *value, size_t len) { \
+    devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *pObj =                         \
+        (devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE> *) drvPvt;                 \
+    pObj->interruptCallback(pasynUser, value, len);                                        \
+}                                                                                          \
+static long INIT_FUNC(dbCommon *pr) {                                                      \
+    new devAsynXXXArray<REC, INTERFACE, INTERRUPT, EPICS_TYPE>                             \
+                        (pr, &((REC *)pr)->LINK, SIGNED_TYPE, UNSIGNED_TYPE, IS_OUTPUT,    \
+                        INTERFACE_NAME, QRCB_FUNC, INTCB_FUNC);                            \
+    return 0;                                                                              \
+}                                                                                          \
+analogDset DSET = {6, 0, 0, INIT_FUNC, GET_INFO_FUNC, PROC_FUNC, 0};                       \
+epicsExportAddress(dset, DSET);
+
+// 8-bit integer arrays
+MAKE_DEVSUP(asynInt8ArrayWfIn, waveformRecord, inp, asynInt8Array, "asynInt8Array", interruptCallbackInt8Array, epicsInt8,
```

Comment:
With some target configuration, `epicsExportAddress()` must be wrapped in `extern "C"`.  The simplest way to do this would be to enclose all of the `MAKE_DEVSUP()` expansions.  eg. `extern "C" { MAKE_DEVSUP(...); ... }`.